### PR TITLE
implements menu sorting by user-provided function

### PR DESCRIPTION
**Pull request by vkichline**
This implements menu sorting in a flexible way, as discussed in [Issue #68](https://github.com/ropg/M5ez/issues/68). The sorting function can be applied before items are added, which keeps the menu sorted on each insert, or after items are inserted if performance is important.

Only one public method is added to ezMenu, which is entirely optional:
`void setSortFunction(bool (*sortFunction)(const char* s1, const char* s2))`.
An example program is included which demonstrates ascending order, descending order, case sensitivity, and caption awareness.

Implementation focuses on reliability rather than performance, so the menu is resorted after every insert. A binary search of the vector and a targeted insert would be faster, but it seems to me that practical menus are limited to a few dozen entries, and the improvement would be more complex but of little detectable value. In cases where a large menu must be sorted, the user can simply fill the menu and then set the sort function, resulting in maximum performance.

Docs and keywords have been augmented in a style consistent with current usage.

Impact to existing code using M5ez should be zero.

Rop, I found a few typo's in the README.md along the way, mostly reversed or dropped letters or trailing spaces, and fixed those as incidentals. If you like, I can break those out into a separate pull request.

### CODE REVIEW

**Comment by paultech** [comment]
This all looks good to me but can we break out the PR's?

- Sorted menus
- Fix preference loss (Closes #73)
- Typo fixes

One question, however; why was ezClock disabled? Judging by comment it was for dev branch? If intentional I think we'd want to discuss first.

**Comment by vkichline** [comment]
I’d be happy to break these out.
I’m not clear on disabling ezClock, was that directed at me? I had no such intention...

Should I close the PR and submit new ones? I’m afraid I’m not too familiar with best practices for OSS. I worked at Microsoft 23 years, and our division had just moved to Git the year I left.

Is there a best practices description somewhere you follow for submissions, reviews, etc.? I am happy to read up.

I am quite concerned about testing. Is there a system in place? It’s certainly a complex enough library to make introducing unintended side effects possible.

Van

File: src/M5ez.h [hunk 1]
```diff
@@ -26,7 +26,7 @@
 #define M5EZ_BACKLIGHT
 
 // Compile in ezTime and create a settings menu for clock display
-#define M5EZ_CLOCK
+// #define M5EZ_CLOCK
```
Comment (paultech) [inline src/M5ez.h:29]: Here @vkichline

**Comment by paultech** [comment]
I'm fan of git flow for best practices. 

[git flow]( https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow)

Provides a branching model that's easy to work with: 

```
Branch name for production releases: [master]
Branch name for "next release" development: [develop]
How to name your supporting branch prefixes?
Feature branches? [feature/]
Release branches? [release/]
Hotfix branches? [hotfix/]
Support branches? [support/]
```
Ideally, one PR per feature change as it provides accountability and ability to code-review.

Unit-tests are going to be high up on our priority list to implement. We do have enough codebase to justify them and we have a few different options to go with.  [AUnit](https://github.com/bxparks/AUnit) as an example.

**Comment by ropg** [comment]
> I am quite concerned about testing. Is there a system in place? It’s certainly a complex enough library to make introducing unintended side effects possible.


We use HFTB for testing ("Hope For The Best"). Yes, something more is in order. I have no experience with automated testing, but it sure does sound cool as a concept.

Rop

**Comment by vkichline** [comment]
Just getting organized and re-reading everything.
I had completely forgotten doing that and certainly didn’t mean to submit it.
I’ll do the PR over, broken up. Just not sure how to do that yet…

**Pull request closed without merging**